File: src/main/java/com/Backend/music_app/dto/request/PlaylistCreateRequest.java
package com.Backend.music_app.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class PlaylistCreateRequest {
    String title;
    @JsonProperty("isPublic")
    boolean isPublic;
    @JsonProperty("coverUrl")
    String coverUrl;
    @JsonProperty("isSystem")
    boolean isSystem;;

}

File: lib/data/createRequest/PlaylistCreate_request.dart
class PlaylistCreateRequest
{
  final String title;
  final bool isPublic;
  final String coverUrl;
  final bool isSystem;

  PlaylistCreateRequest({
    required this.title,
    required this.coverUrl,
    required this.isPublic,
    required this.isSystem,
  });

  factory PlaylistCreateRequest.fromJson(Map<String, dynamic> json) {
    return PlaylistCreateRequest(
      title: json['title'] ?? '',
      coverUrl: json['coverUrl'] ?? '',
      isPublic: json['isPublic'] ?? false,
      isSystem: json['isSystem'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'isPublic': isPublic,
      'coverUrl': coverUrl,
      'isSystem': isSystem,
    };
  }
}